digitalocean/database/resource_database_cluster.go

```diff
@@ -524,6 +557,13 @@ func resourceDigitalOceanDatabaseClusterRead(ctx context.Context, d *schema.Reso
 	if err != nil {
 		return diag.Errorf("Error setting connection info for database cluster: %s", err)
 	}
+
+	if database.EngineSlug == opensearchEngineSlug {
```

Comment:
not sure we need to check opensearch specifically, we have a check in the `setUIConnectionInfo` for `database.UIConnection != nil`. Just so it can be general purpose for other ui connection details for other engine types.